lib/src/handler_client.dart

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'base_client.dart';
+import 'request.dart';
+import 'response.dart';
+
+typedef Future<Response> Handler(Request request);
+typedef void CloseHandler();
```

Comment:
Yeah, that sounds fine—1.24 will probably be out before this is ready for release.